Formula/s/spoofdpi.rb

```diff
@@ -0,0 +1,70 @@
+class Spoofdpi < Formula
+  desc "Simple and fast anti-censorship tool written in Go"
+  homepage "https://github.com/xvzc/SpoofDPI"
+  url "https://github.com/xvzc/SpoofDPI/archive/refs/tags/0.10.10.tar.gz"
+  sha256 "2cc1cfe8f05246edf586430d25382a29f8808fd37e097f1bb9e89bcb3f3c3fa0"
+  license "Apache-2.0"
+  head "https://github.com/xvzc/SpoofDPI.git", branch: "main"
+
+  depends_on "go" => :build
+  uses_from_macos "curl" => :test
+
+  def install
+    ohai "Starting the build..."
+    system "go", "build", *std_go_args(ldflags: "-s -w"), "./cmd/spoof-dpi"
+    ohai "Build completed successfully"
+  end
+
+  test do
+    port = free_port
+    ohai "Got port: #{port}"
+
+    pid=start_server(port)
+    ohai "Got PID: #{pid}"
+
+    test_can_connect(port)
+    test_set_proxy_on_macos(port) if OS.mac?
+
+    Process.kill("SIGTERM", pid)
+  end
+
+  def start_server(port)
+    ohai "Starting SpoofDPI in the background with -port=#{port}..."
+    pid = fork do
+      exec "#{bin}/spoofdpi -port #{port}"
+    end
+    sleep 1
+    pid
+  end
+
+  def test_can_connect(port)
+    # Check that a listener is created successfully by connecting to it via telnet.
+    #
+    # "nothing" is an invalid option, but curl will process it
+    # only after it succeeds at establishing a connection,
+    # then it will close it, due to the option, and return exit code 49.
+
+    ohai "Checking availability of the server..."
+    shell_output("curl -s --connect-timeout 1 --telnet-option nothing 'telnet://127.0.01:#{port}' > /dev/null", 49)
+    ohai "Success"
+  end
+
+  def test_set_proxy_on_macos(port)
+    # Check that the server set correct system proxy settings.
+
+    ohai "Checking configuration of system proxy settings..."
+    output = shell_output("scutil --proxy | grep HTTP")
+    proxy = lines_to_hash(output, " : ")
+
+    ["HTTP", "HTTPS"].each do |proto|
+      assert_equal "1", proxy["#{proto}Enable"]
+      assert_equal port.to_s, proxy["#{proto}Port"]
+      assert_equal "127.0.0.1", proxy["#{proto}Proxy"]
+    end
+    ohai "Success"
+  end
+
+  def lines_to_hash(lines, sep)
+    lines.each_line.to_h { |l| l.strip.split(sep, 2) }
+  end
```

Comment:
Please just collapse this into the test block, there is no need to make this reusable.

Reply:
Done